dev/com.ibm.ws.security.spnego.fat.common/fat/src/com/ibm/ws/security/spnego/fat/config/ApacheKDCforSPNEGO.java

```diff
@@ -0,0 +1,172 @@
+import org.apache.directory.api.ldap.model.entry.Entry;
+import org.apache.directory.server.kerberos.shared.keytab.Keytab;
+import org.apache.directory.server.kerberos.shared.keytab.KeytabEntry;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.log.Log;
+import com.ibm.ws.security.wim.adapter.ldap.fat.krb5.ApacheDSandKDC;
+
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.utils.FileUtils;
+
+@RunWith(FATRunner.class)
+public class ApacheKDCforSPNEGO extends ApacheDSandKDC {
+
+    private static final Class<?> c = ApacheKDCforSPNEGO.class;
+
+    public static String SPN;
+
+    public static String SPN_PASSWORD = "httppwd";
+
+    private static String spnegoUserDN;
+
+    public static String canonicalHostname = "canhostname1";
+
+    private static String SpnegoSPNKeytabFile = null;
+
+    //ADDED ON
+    //KDC Users
+    public static String KRB5_USER = bindUserName;
+    public static String KRB5_USER_PWD = bindPassword;
+
+    public static String KRB5_USER1 = KRB5_USER; //"user1";
+    public static String KRB5_USER1_PWD = KRB5_USER_PWD; //"user1pwd";
+    public static String KRB5_USER2 = KRB5_USER; //"user2";
+    public static String KRB5_USER2_PWD = KRB5_USER_PWD; //"user2pwd";
+
+    public static String COMMON_TOKEN_USER = KRB5_USER1;
+    public static String COMMON_TOKEN_USER_PWD = KRB5_USER1_PWD;
+    public static boolean COMMON_TOKEN_USER_IS_EMPLOYEE = true;
+    public static boolean COMMON_TOKEN_USER_IS_MANAGER = false;
+
+    public static String COMMON_SPNEGO_TOKEN = null;
+    public static String KEYTAB_FILE_LOCATION = null;
+    public static long COMMON_TOKEN_CREATION_DATE = 0;
+    public static final double TOKEN_REFRESH_LIFETIME_SECONDS = 180;
+    public static boolean LOCALHOST_DEFAULT_IP_ADDRESS = true; //127.0.0.1       localhost
+
+    public static boolean IBM_JDK_V8_LOWER = false;
+    public static boolean OTHER_SUPPORT_JDKS = false;
+    public static boolean SUN_ORACLE_JDK_V8_HIGHER = false;
+    public static boolean IBM_HYBRID_JDK = false;
+
+    public static String serverShortHostName = "shortHostName";
+
+    public static String KDC_REALM = DOMAIN;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+
+        WHICH_FAT = "SPNEGO";
+
+        bindUserName = "user1";
+        bindPassword = "user1pwd";
+        bindPrincipalName = bindUserName + "@" + DOMAIN;
+        KRB5_USER = bindUserName;
+        KRB5_USER_PWD = bindPassword;
+        KRB5_USER1 = KRB5_USER;
+        KRB5_USER1_PWD = KRB5_USER_PWD;
+        COMMON_TOKEN_USER = KRB5_USER1;
+
+        KRB5_USER2 = "user2";
+        KRB5_USER2_PWD = "user2pwd";
+
+        spnegoUserDN = "uid=" + canonicalHostname + "," + BASE_DN;
+
+        SPN = "HTTP/" + canonicalHostname + "@" + DOMAIN;
+
+        setupService();
+
+        ApacheKDCCommonTest.setGlobalLoggingLevel(Level.INFO);
```

Comment:
Why do we configure the test / logging here? Isn't this just a utility that the test uses? Maybe I am not following the hierarchy of classes.

Reply:
Add comment: ApacheCommons logging adds a lot of extra debug that is not helpful and clutter